common/djangoapps/third_party_auth/samlproviderdata/views.py

```diff
@@ -0,0 +1,64 @@
+"""
+    Viewset for auth/saml/v0/samlproviderdata
+"""
+
+from django.http import Http404
+from django.shortcuts import get_object_or_404
+from edx_rbac.mixins import PermissionRequiredMixin
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from enterprise.models import EnterpriseCustomerIdentityProvider
+from rest_framework import permissions, viewsets
+from rest_framework.authentication import SessionAuthentication
+
+from ..models import SAMLProviderConfig, SAMLProviderData
+from .serializers import SAMLProviderDataSerializer
+
+
+class SAMLProviderDataMixin(object):
+    authentication_classes = [JwtAuthentication, SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = SAMLProviderDataSerializer
+
+
+class SAMLProviderDataViewSet(PermissionRequiredMixin, SAMLProviderDataMixin, viewsets.ModelViewSet):
+    """
+    A View to handle SAMLProviderData CRUD.
+    Uses the edx-rbac mixin PermissionRequiredMixin to apply enterprise authorization
+
+    Usage:
+        [HttpVerb] /auth/saml/v0/providerdata/
+    """
+    permission_required = 'enterprise.can_access_admin_dashboard'
+
+    def get_queryset(self):
+        """
+        Find and return the matching providerid for the given enterprise uuid
+        Note: There is no direct association between samlproviderdata and enterprisecustomer.
+        So we make that association in code via samlproviderdata > samlproviderconfig ( via entity_id )
+        then, we fetch enterprisecustomer via samlproviderconfig > enterprisecustomer ( via association table )
+        """
+        if self.requested_enterprise_uuid is None:
+            raise Http404('Required enterprise_customer_uuid is missing')
+        enterprise_customer_idp = get_object_or_404(EnterpriseCustomerIdentityProvider,
+            enterprise_customer__uuid=self.requested_enterprise_uuid)
+        saml_provider = SAMLProviderConfig.objects.get(pk=enterprise_customer_idp.provider_id)
+        return SAMLProviderData.objects.filter(entity_id=saml_provider.entity_id)
+
+    @property
+    def requested_enterprise_uuid(self):
+        """
+        The enterprise customer uuid from request params or post body
+        """
+        if self.request.method in ('POST', 'PATCH', 'DELETE'):
+            uuid_str = self.request.POST.get('enterprise_customer_uuid')
```

Comment:
Not sure if you'll have a POST body on PATCH/DELETE, unclear what Django'll have. I would expect us to not need to resolve these objects on PUT/PATCH/DELETE by `enterprise_customer_uuid` depending on our access pattern from our front end, I'd expect us to just operate off of the primary key in those scenarios (see comment above)

Reply:
yeah I was surprised too, but apparently they all have post body.. there is no request.PATCH or request.DELETE  it seems ... 

on the enterprise_customer_uuid piece, as per prior chats, looks like we decided to always require this in the body unless it's a GET? we are using it as a kind of authorization since currently this has to match the decoded cookie uuid for RBAC, I think